templates/hpaV1.yaml

```diff
@@ -0,0 +1,21 @@
+{{- if .Values.autoscaling.enabled }}
+{{- if eq .Values.autoscaling.apiVersion "v1" -}}
```

Comment:
Did you know there is a `.Capabilities.APIVersions` collection available here that can be used to query the target cluster's API version support? It could alleviate the need for an `autoscaling.apiVersion` value -- opportunisticslly use v2 if supported, else the chart could fallback to v1.

Reply:
I was in dilemma regarding this. I went with approach to enable user selecting version, giving him possibility to decide when he would like to update _HPA_ independent from kubernetes upgrades. Because using `.Capabilities.APIVersions` basically forces the user to upgrade HPA when updating kubernetes cluster version. But on other hand, these changes are always with high-impacting so some migration activities are always in order. Also the same "migration flow" is for _Ingress_. 

I have removed `autoscaling.apiVersion` variable and use `.Capabilities.APIVersions` for _HPA_ resources. I also renamed _HPA_ file for V2 to `hpa.yaml` so when `autoscaling/v1` is truly deprecated, it makes things easier. As suggested it will use `autoscaling/v2` if available else fallback to `autoscaling/v1` (provided `autoscaling.enabled` is set to `true`)